docs/rules/strict.md

```diff
@@ -139,6 +139,22 @@ function foo() {
 }());
 ```
 
+```js
+/*eslint strict: ["error", "function"]*/
+/*eslint-env es6*/
+
+// Illegal "use strict" directive in function with non-simple parameter list.
+// This is a syntax error since ES2016.
+function foo(a = 1) {
+    "use strict";
+}
+
+// We cannot write "use strict" directive in this function.
+// So we have to wrap this function with a function with "use strict" directive.
```

Comment:
Yeah, actually, the warning message of such this function is "Wrap this function in a function with 'use strict' directive." for people which don't read documents.